shared/util.go

```diff
@@ -651,9 +651,10 @@ func (r BytesReadCloser) Close() error {
 	return nil
 }
 
-// IsSnapshot returns true if a given name contains the snapshot delimiter.
+// IsSnapshot returns true if a given name contains a snapshot delimiter not at the end.
 func IsSnapshot(name string) bool {
```

Comment:
Probably we should centralise the "is snapshot/extract snapshot parts" into api.GetParentAndSnapshotName instead.